simple/msg_setname.c

```diff
@@ -0,0 +1,647 @@
+	ret = alloc_ep_res(fi);
+	if (ret)
+		goto err5;
+
+	/* Close the passive endpoint that we "stole" the source address
+	 * from */
+	ret = fi_close(&pep->fid);
+	if (ret)
+		goto err5;
+
+	ret = check_address(&ep->fid, "fi_endpoint (ep)");
+	if (ret)
+		goto err5;
+
+	ret = bind_ep_res();
+	if (ret)
+		goto err6;
+
+	/* Connect to server */
+	ret = fi_connect(ep, fi->dest_addr, NULL, 0);
+	if (ret) {
+		FT_PRINTERR("fi_connect", ret);
+		goto err6;
+	}
+
+	/* Wait for the connection to be established */
+	rd = fi_eq_sread(cmeq, &event, &entry, sizeof entry, -1, 0);
+	if (rd != sizeof entry) {
+		FT_PRINTERR("fi_eq_sread", rd);
+		return (int) rd;
+	}
+
+	if (event != FI_CONNECTED || entry.fid != &ep->fid) {
+		FT_ERR("Unexpected CM event %d fid %p (ep %p)\n", event, entry.fid, ep);
+		ret = -FI_EOTHER;
+		goto err6;
+	}
+
+	ret = check_address(&ep->fid, "connect");
+	if (ret) {
+		goto err6;
+	}
+
+	fi_freeinfo(fi);
+	return 0;
+
+err6:
+	free_ep_res();
+err5:
+	fi_close(&cmeq->fid);
+err4:
+	fi_close(&dom->fid);
+err2:
+	fi_freeinfo(fi);
+err0:
+	return ret;
+}
+
+static int send_recv()
+{
+	struct fi_cq_entry comp;
+	int ret;
+
+	if (opts.dst_addr) {
+		/* Client */
+		fprintf(stdout, "Posting a send...\n");
+		sprintf(buf, "Hello World!"); 
+		ret = fi_send(ep, buf, sizeof("Hello World!"), fi_mr_desc(mr), 0, buf);
+		if (ret) {
+			FT_PRINTERR("fi_send", ret);
+			return ret;
+		}
+
+		/* Read send queue */
+		do {
+			ret = fi_cq_read(scq, &comp, 1);
+			if (ret < 0 && ret != -FI_EAGAIN) {
+				FT_PRINTERR("fi_cq_read", ret);
+				return ret;
+			}
+		} while (ret == -FI_EAGAIN);
+
+		fprintf(stdout, "Send completion received\n");
+	} else {
+		/* Server */
+		fprintf(stdout, "Posting a recv...\n");
+		ret = fi_recv(ep, buf, buffer_size, fi_mr_desc(mr), 0, buf);
+		if (ret) {
+			FT_PRINTERR("fi_recv", ret);
+			return ret;
+		}
+
+		/* Read recv queue */
+		fprintf(stdout, "Waiting for client...\n");
+		do {
+			ret = fi_cq_read(rcq, &comp, 1);
+			if (ret < 0 && ret != -FI_EAGAIN) {
+				FT_PRINTERR("fi_cq_read", ret);
+				return ret;
+			}
+		} while (ret == -FI_EAGAIN);
+
+		fprintf(stdout, "Received data from client: %s\n", (char *)buf);
+	}
+
+	return 0;
+}
+
+static int setup_handle(void)
+{
+	static char buf[BUFSIZ];
+	struct addrinfo *ai, aihints;
+	struct fi_info *fi;
+	int ret;
+
+	memset(&aihints, 0, sizeof aihints);
+	aihints.ai_flags = AI_PASSIVE;
+	ret = getaddrinfo(opts.src_addr, opts.src_port, &aihints, &ai);
+	if (ret == EAI_SYSTEM) {
+		FT_PRINTERR("getaddrinfo", -ret);
+		return -ret;
+	} else if (ret) {
+		FT_ERR("getaddrinfo: %s\n", gai_strerror(ret));
+		return -FI_ENODATA;
+	}
+
+	switch (ai->ai_family) {
+	case AF_INET:
+		hints->addr_format = FI_SOCKADDR_IN;
+		break;
+	case AF_INET6:
+		hints->addr_format = FI_SOCKADDR_IN6;
+		break;
+	case AF_IB:
+		hints->addr_format = FI_SOCKADDR_IB;
+		break;
+	}
+
+	/* Get fabric info */
+	ret = fi_getinfo(FT_FIVERSION, NULL, NULL, 0, hints, &fi);
```

Comment:
> @shefty <https://github.com/shefty>  Is fi_getinfo(FT_FIVERSION, NULL,
> NULL, FI_SOURCE, hints, &fi); with src_addr and dest_addr in hints NULL'd
> out a sane/correct thing to do in this case?

Adding the FI_SOURCE flag doesn't help any, since you haven't specified a source address anywhere. So, I would vote no, it doesn't really make sense.